boa_engine/src/object/builtins/jstypedarray.rs

```diff
@@ -273,6 +273,84 @@ impl JsTypedArray {
         )
     }
 
+    /// Calls `TypedArray.prototype.findIndex()`.
+    #[inline]
+    pub fn find_index(
+        &self,
+        predicate: JsFunction,
+        this_arg: Option<JsValue>,
+        context: &mut Context,
+    ) -> JsResult<Option<usize>> {
```

Comment:
This should return `u64`, since JS considers numbers below 2^53 - 1 as valid indices. I don't think we ensure this in `JsArray` so I'll open an issue for that.

Reply:
i think find methods do this because they might return -1

Reply:
Could `Option::None` serve as an alternative to -1? How would `Option::None` be translated into the JavaScript world?

Reply:
No, I meant returning `None` if `index` is negative, and converting the `f64` to `u64` otherwise.

Reply:
did it, but I'm wondering: the function api should return -1 if the element isn't found, how does `None` appear to the end user in JavaScript?

Reply:
I don't think we have a direct conversion from `Option<T>` to `JsValue`, but if we do, it would probably map as `Some(T) -> JsValue::from(T)` and `None -> JsValue::Undefined`

Reply:
Do you think we should see this as a problem now, or should we wait for later?

Reply:
I don't think that blocks adding this? This will be exclusively a Rust API. Implementors can handle an `Option` however they want to, if they need to somehow "expose" the result to JS again.